apps/src/blockly/addons/functionEditor.js

```diff
@@ -27,3 +27,48 @@ export default class FunctionEditor {
   // TODO
   openWithNewFunction() {}
 }
+
+export function procedureFlyoutCallback(workspace) {
+  const xmlList = [];
+  if (Blockly.Blocks['procedures_defnoreturn']) {
+    // <block type="procedures_defnoreturn" gap="16">
+    //     <field name="NAME">do something</field>
+    // </block>
+    const block = Blockly.utils.xml.createElement('block');
+    block.setAttribute('type', 'procedures_defnoreturn');
+    block.setAttribute('gap', 16);
+    const nameField = Blockly.utils.xml.createElement('field');
+    nameField.setAttribute('name', 'NAME');
+    nameField.appendChild(
+      Blockly.utils.xml.createTextNode(
+        Blockly.Msg['PROCEDURES_DEFNORETURN_PROCEDURE']
+      )
+    );
+    block.appendChild(nameField);
+    xmlList.push(block);
+
+    if (xmlList.length) {
+      // Add slightly larger gap between system blocks and user calls.
+      xmlList[xmlList.length - 1].setAttribute('gap', 24);
+    }
+
+    const allProcedureCalls = Blockly.Procedures.allProcedures(workspace)[0];
```

Comment:
Could you add a comment describing what this is doing? Having a hard time following it without much context.

Reply:
Would this work be technically part of the function modal editor? I'm unsure if I'm reading it correctly, but it seems like it is modifying Functions options from the toolbox menu.

Reply:
@bencodeorg We're doing this so that we can auto-populate the flyout category with call blocks for each function (procedure) defined in the workspace. 
https://developers.google.com/blockly/reference/js/blockly.procedures_namespace.allprocedures_1_function.md
<img width="470" alt="image" src="https://user-images.githubusercontent.com/43474485/195866127-ecb4d45f-beb1-4299-b637-dcf37a3792a9.png">

@fisher-alice We're not addressing the modal editor yet. These changes just update the function blocks in Google Blockly levels to better align them to how they appear in CDO Blockly levels _that do not use the modal editor_. 
My branch name was a fair bit too ambitious.

Reply:
I agree. I think we could probably add eyes tests here that shows that the Functions category before before and after adding a new function definition.

Reply:
Thanks Mike - I was just wondering if the scope of this work belonged in this functionEditor file.

Reply:
Discussed on Slack - Mike shared with me where this customization to the flyout categories is currently done in @code-dot-org/blockly. [`core/utils/procedures.js`](https://github.com/code-dot-org/blockly/blob/main/core/utils/procedures.js)

Reply:
@mikeharv while I agree having some sort of UI/eyes test of what happens to the Functions category before/after adding new function definitions, the tests I'm envisioning here would be more basic.

eg, does this function return the expected XML given a workspace with a user-defined procedure? How about one with 0-many args?

It also helps with review, b/c you can see the expected outputs of a given function.

All of that said, I'm not sure that's a reasonable standard right now, since almost none of the functionality in our "addons" directory is unit tested.